controllers/serviceinstance_controller_test.go

```diff
@@ -684,23 +662,11 @@ var _ = Describe("ServiceInstance controller", func() {
 			})
 
 			When("delete in SM fails", func() {
-				JustBeforeEach(func() {
-					fakeClient.DeprovisionReturns("", fmt.Errorf("failed to delete instance"))
-				})
-
-				JustAfterEach(func() {
-					fakeClient.DeprovisionReturns("", nil)
-				})
-
 				It("should not delete the k8s instance and should update the condition", func() {
+					fakeClient.DeprovisionReturns("", fmt.Errorf("failed to delete instance"))
 					deleteInstance(ctx, serviceInstance, false)
-					Eventually(func() bool {
-						err := k8sClient.Get(ctx, defaultLookupKey, serviceInstance)
-						if err != nil {
-							return false
-						}
-						return isFailed(serviceInstance)
-					}, timeout, interval).Should(BeTrue())
+					waitForInstanceToBeFailed(ctx, defaultLookupKey)
```

Comment:
this is too general, you should check the condition contains the message returned from SM and the right status

Reply:
done